pkg/health/membergarbagecollector/membergarbagecollector.go

```diff
@@ -0,0 +1,171 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package membergarbagecollector
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/gardener/etcd-backup-restore/pkg/errors"
+	"github.com/gardener/etcd-backup-restore/pkg/etcdutil"
+	"github.com/gardener/etcd-backup-restore/pkg/miscellaneous"
+	utils "github.com/gardener/etcd-backup-restore/pkg/snapstore"
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	etcdclient "github.com/gardener/etcd-backup-restore/pkg/etcdutil/client"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	podNamespace = "POD_NAMESPACE"
+	podName      = "POD_NAME"
+)
+
+// MGCChecker garbage collects the members from the etcd cluster if its corresponding pod is not present. (Used for unit tests only)
+type MGCChecker interface {
+	RemoveSuperfluousMembers(ctx context.Context) error
+}
+
+// MemberGarbageCollector contains information to regularly cleanup superfluous ETCD members
+type MemberGarbageCollector struct {
+	logger       *logrus.Entry
+	gcTimer      *time.Timer
+	k8sClient    client.Client
+	podNamespace string
+	etcdName     string
+}
+
+func init() {
+	druidv1alpha1.AddToScheme(scheme.Scheme)
+}
+
+// NewMemberGarbageCollector returns the member garbage collect object
+func NewMemberGarbageCollector(logger *logrus.Entry, clientSet client.Client) (*MemberGarbageCollector, error) {
+	if clientSet == nil {
+		return nil, &errors.EtcdError{
+			Message: "nil kubernetes clientset passed, can not create etcd member garbage collector",
+		}
+	}
+	namespace, err := utils.GetEnvVarOrError(podNamespace)
+	if err != nil {
+		logger.Fatalf("POD_NAMESPACE env var not present: %v", err)
+	}
+	podName, err := utils.GetEnvVarOrError(podName)
+	if err != nil {
+		logger.Fatalf("POD_NAME env var not present: %v", err)
+	}
+	return &MemberGarbageCollector{
+		logger:       logger.WithField("actor", "etcd-membergc"),
+		k8sClient:    clientSet,
+		podNamespace: namespace,
+		etcdName:     podName[:strings.LastIndex(podName, "-")], //TODO Evaluate if this will always be sufficient
+	}, nil
+}
+
+// RunMemberGarbageCollectorPeriodically periodically calls the member garbage collector
+func RunMemberGarbageCollectorPeriodically(ctx context.Context, hconfig *brtypes.HealthConfig, logger *logrus.Entry, etcdConfig *brtypes.EtcdConnectionConfig) {
+	clientSet, err := miscellaneous.GetKubernetesClientSetOrError()
+	if err != nil {
+		logger.Errorf("failed to create kubernetes clientset: %v", err)
+		return
+	}
+	mgc, err := NewMemberGarbageCollector(logger, clientSet)
+	if err != nil {
+		logger.Errorf("failed to initialize new etcd member garbage collector: %v", err)
```

Comment:
```suggestion
		mgc.logger.Errorf("failed to initialize new etcd member garbage collector: %v", err)
```

Reply:
Same as above. This error log is printed if there is an error creating the `mgc` object